awscli/clidocs.py

```diff
@@ -716,3 +710,42 @@ def _line_has_tag(self, line):
 
     def doc_subitems_start(self, help_command, **kwargs):
         pass
+
+
+class GlobalOptionsDocumenter:
+    """Documenter used to pre-generate global options docs."""
+
+    def __init__(self, help_command):
+        self._help_command = help_command
+
+    def _remove_multilines(self, s):
+        return re.sub(r'\n+', '\n', s)
+
+    def doc_global_options(self):
+        help_command = self._help_command
+        for arg in help_command.arg_table:
+            argument = help_command.arg_table.get(arg)
+            help_command.doc.write((f"``{argument.cli_name}`` "
+                                    f"({argument.cli_type_name})"))
+            help_command.doc.include_doc_string(argument.documentation)
+            if argument.choices:
+                help_command.doc.writeln(" ")
+                help_command.doc.style.start_ul()
+                for choice in argument.choices:
+                    help_command.doc.style.li(choice)
+                help_command.doc.style.end_ul()
+            help_command.doc.writeln(" ")
```

Comment:
Another thing that I noticed while taking another pass at it is that the option docs are slightly different than the operation options:

1. The docstrings are on a new line
2. The docstrings are indented

I realize this is mainly ported from the original implementation, but at this point it probably makes sense to make them consistent in format between the two different types of options. Playing around with this a bit more. I was able to update it to this to do the job:
```python
            help_command.doc.writeln(
                f"``{argument.cli_name}`` ({argument.cli_type_name})")
            help_command.doc.style.indent()
            help_command.doc.style.new_paragraph()
            help_command.doc.include_doc_string(argument.documentation)
            if argument.choices:
                help_command.doc.style.start_ul()
                for choice in argument.choices:
                    help_command.doc.style.li(choice)
                help_command.doc.style.end_ul()
            help_command.doc.style.dedent()
            help_command.doc.style.new_paragraph()
```
I think there may be a couple places where there are more newlines than needed but that's fine.